src/AspNet.Security.OAuth.Amazon/AmazonAuthenticationHandler.cs

```diff
@@ -4,31 +4,37 @@
  * for more information concerning the license and the contributors participating to this project.
  */
 
+using System.Linq;
 using System.Net.Http;
 using System.Net.Http.Headers;
 using System.Security.Claims;
+using System.Text.Encodings.Web;
 using System.Threading.Tasks;
-using AspNet.Security.OAuth.Extensions;
 using JetBrains.Annotations;
 using Microsoft.AspNetCore.Authentication;
 using Microsoft.AspNetCore.Authentication.OAuth;
-using Microsoft.AspNetCore.Http.Authentication;
 using Microsoft.AspNetCore.WebUtilities;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
 using Newtonsoft.Json.Linq;
 
 namespace AspNet.Security.OAuth.Amazon
 {
     public class AmazonAuthenticationHandler : OAuthHandler<AmazonAuthenticationOptions>
     {
-        public AmazonAuthenticationHandler([NotNull] HttpClient client)
-            : base(client)
+        public AmazonAuthenticationHandler(
+            [NotNull] IOptionsMonitor<AmazonAuthenticationOptions> options,
+            [NotNull] ILoggerFactory logger,
+            [NotNull] UrlEncoder encoder,
+            [NotNull] ISystemClock clock)
+            : base(options, logger, encoder, clock)
         {
         }
 
         protected override async Task<AuthenticationTicket> CreateTicketAsync([NotNull] ClaimsIdentity identity,
             [NotNull] AuthenticationProperties properties, [NotNull] OAuthTokenResponse tokens)
         {
-            var endpoint = Options.UserInformationEndpoint;
+            string endpoint = QueryHelpers.AddQueryString(Options.UserInformationEndpoint, "access_token", tokens.AccessToken);
```

Comment:
Why this change? We don't want functional changes as part of the 2.0 port.

Reply:
This comes from how I originally implemented Amazon authentication in my own project before I did the initial PR to add it here: [link](https://github.com/martincostello/alexa-london-travel-site/blob/28b5adea435fdaed519add79e4f4dff1fab30889/src/LondonTravel.Site/Identity/Amazon/AmazonHandler.cs#L33)

Reply:
It's always preferable to flow the access token via the `Authorization` header, if possible (which is the approach recommended by the OAuth2 bearer spec).

If Amazon deprecated or removed that, please open a new ticket so we can change the 1.x provider too.